components/PoeModal.tsx

```diff
@@ -758,6 +823,23 @@ const PoeModal = () => {
 
                           setTimeoutAndHash(timeoutAndHash);
 
+                          if (signImmediatelyFlag) {
+                            const storage = toStorage(
+                              versioned.version,
+                              await cc.storage(),
+                              BigNumber(0)
+                            );
+                            const proposal_id = storage.proposal_counter;
+                            timeoutAndHash = await versioned.signProposal(
+                              cc,
+                              state.connection,
+                              proposal_id,
```

Comment:
Same than before for the styling. Also code looks like it's the same, can it be refactored into a common function?

Reply:
Done, about the code repeated, probably I will implement this in the contract handler